### ISSUE TITLE
Animation just won't start.

### ISSUE THREAD

**Issue by pepo972** [jlmakes/scrollreveal#338]
Hello,
First, thanks for the awesome project and the chance to use it! But now on the question.

I'm trying to use it on an image, but it just won't trigger the animation. In my web page, the image is still there but invisible. Here is part of my code:

```html
<img class="tags" src="tags.png" alt="HTML Tags Example">
```

```js
window.sr = new ScrollReveal()
sr.reveal('.tags', { origin: 'bottom', distance: '20px', duration: 2000, })
```

Where could my problem be? I've just started learning JS, so the thing could be silly mistakes, but couldn't fix it on myself. Thank you in advance!

**Comment by jlmakes** [comment]
Glad to hear you’re learning JavaScript!

There doesn’t appear to be anything wrong with the way you’ve used ScrollReveal.
In fact, I created a [JSBin with your code](http://jsbin.com/varijaw/edit?html,output), and everything works.

*(I did change the image source to a placeholder image for demo purposes.)*

It would seem the problem lies elsewhere. If you can recreate the problem using JSBin, I will gladly take a look at it, but I’m not sure what else to suggest. 🤔

**Comment by PeterShaggyNoble** [comment]
Are you sure SR is the problem and it's not an error elsewhere in your JavaScript preventing it from running? Are there any errors reported in your browser's console?

**Comment by pepo972** [comment]
I get that error in the browser console:

> Uncaught ReferenceError: ScrollReveal is not defined
    at script.js:27

Would the problem be somewhere from the the way I'm importing the libraries? Because I need them on locally, not through link.

This is how I am doing it:
`<script src="jquery-3.2.1.min.js"></script>
	<script type="text/javascript" src="script.js"></script>
	<script src="scrollreveal.min.js"></script>
	<script type="text/javascript" src="script.js"></script>`

The HTML is clear, it's the same like in the example, I don't think that's the problem and the JS for the SR it's the same too.

**Comment by PeterShaggyNoble** [comment]
You're including `script.js` twice, once before `scrollreveal.min.js` and once after it. I'm guessing that the code from your original post appears somewhere in the `script.js`. If that's the case then the problem is that, by including that file before SR, it's making a call to an object that doesn't yet exist so you should be able to solve your problem simply by removing the first `<script type="text/javascript" src="script.js"></script>`.

**Comment by pepo972** [comment]
Did it. Still doesn't want to work... The thing is that when I try to replicate it in the JSBin it's working:
[http://jsbin.com/focosumira/edit?html,js,output](url)
But in my code it is not. The only difference is that I try to add the library locally, but even if I try with link, it's still not showing animation. This is how I tried with the link: (even with commenting the other scripts, nothing changes on the animation):
`<script src="jquery-3.2.1.min.js"></script>
	
	<script src="https://unpkg.com/scrollreveal/dist/scrollreveal.min.js"></script>
	<script type="text/javascript" src="script.js"></script>`

**Comment by jlmakes** [comment]
Well @pepo972, as already mentioned, the order is very important.

**ScrollReveal should be included in your page, before you attempt to use it.** The error message you’re getting suggests that you are attempting to use `new ScrollReveal()` before the ScrollReveal script has been recognized by the browser.

What’s confusing is, your latest post shows that you are importing ScrollReveal *before* your `scripts.js` file (where I assume you are attempting to use ScrollReveal). Furthermore, you’ve already seen for yourself that in an isolated environment (JSBin) that everything works fine.

Are you using any tools like CodeKit for example, that may be affecting the order of your scripts before they are served by your browser?

**Comment by pepo972** [comment]
No, I'm not using any other tools. I've tried to make a local copy of the JSBin one that is working online and the strange thing (again) is that it's still not working, even with the most basic setup. Tried on Firefox and Opera too, if somehow there could be a problem with my Chrome browser, still doesn't want to fire up. Very strange that.

**Comment by pepo972** [comment]
Someone still wanting to help? Been stuck on that for so long, can't get it out of my head...

**Comment by jlmakes** [comment]
When you type `ScrollReveal` into your browser’s console, does it return the constructor function or an error message?

**Comment by pepo972** [comment]
@jlmakes Doesn't show anything at all.

**Comment by jlmakes** [comment]
That can't be right. It will either show the constructor function as demonstrated above, or return an error message:

![sr2](https://cloud.githubusercontent.com/assets/2044842/25727169/0eec6954-3128-11e7-9907-8fe6cb92572e.gif)

**Comment by pepo972** [comment]
My mistake! It's showing the constructor function. 

`function e(n){return"undefined"==typeof this||Object.getPrototypeOf(this)!==e.prototype?new e(n):(O=this,O.version="3.3.4",O.tools=new E,O.isSupported()?(O.tools.extend(O.defaults,n||{}),O.defaults.container=…`

**Comment by jlmakes** [comment]
So @pepo972, that means you are correctly loading ScrollReveal into the page. That means if you are receiving this error message from your code:

> I get that error in the browser console:
> `Uncaught ReferenceError: ScrollReveal is not defined at script.js:27`

The problem is that you are attempting to use `ScrollReveal` before the library has been loaded. The probably means your personal JavaScript is included on the page _before_ the `scrollreveal.min.js` script tag.

Can you again share how you are including scripts on your page?

**Comment by pepo972** [comment]
Like that:
`<script src="scrollreveal.min.js"></script>
	<script src="jquery-3.2.1.min.js"></script>
	<script type="text/javascript" src="script.js"></script>`
And I have them both files in my src folder:

http://i.imgur.com/JxUg462.png

**Comment by jlmakes** [comment]
Well the source order appears to be fine.

Can you upload the contents of your `script.js` to a JSBin or something?

**Comment by pepo972** [comment]
Here you are: https://jsbin.com/pamukokume/edit?js,output

**Comment by jlmakes** [comment]
Well you are not alone in being confused—I don't see why ScrollReveal would be undefined. Have you tried removing jQuery and everything from `script.js` except lines 27 and 29?

At this point, the process of elimination is what I would experiment with. It would be helpful to simplify your project until ScrollReveal works, so the point of failure could be identified.

**Comment by pepo972** [comment]
Did it, removed every js except the scroll reveal, removed 99% of my CSS and most of my HTML, made in bare bones and it's still not working... This is what my code looks like right now: 

https://jsbin.com/zayezokika/edit?html,css,js,output 

And the funny part is that is working on jsbin, again, but still won't fire in my site. I really don't know what the **** that is and why is happening...

**Comment by jlmakes** [comment]
Not sure what else to suggest 😕

**Comment by pepo972** [comment]
@jlmakes  Sad story... I don't know how it would sound, but It's my last chance I think. :D Now, I am more curious, why it isn't working, where is the problem, rather than just making it work and forget. Would it be insolent from me if I can ask you (only if you have a bit of free time), just to send you an archive of my code, to check it, if it will work for you? If not, it's alright, you've done enough! Thank you!

**Comment by jlmakes** [comment]
If you create a public repository with the broken project, I will clone it and confirm whether or not it works for me.

**Comment by pepo972** [comment]
https://github.com/pepo972/Broken-animation - Please, tell me that I did it somewhat right, I'm new around here. Otherwise, if everything is fine, this is my whole folder. 
P.S. Please, don't overlook on the content, it's just a small uni test project.

**Comment by kimmenbert** [comment]
It works if you put
`window.sr = new ScrollReveal();
sr.reveal('.tags', { origin: 'bottom', distance: '20px', duration: 2000, });`

inside `$(document).ready(function() {`

I´m no JavaScript expert, but might this have something to do with scroll reveal trying to locate the element before the element is present?

**Comment by jlmakes** [comment]
@pepo972 I thought we were troubleshooting the `is not defined` error message, but I didn’t experience any error messages... anyhow @kimmenbert is correct—if you have JavaScript in the `<head>` of your document, the `<body>` content is not necessarily available. 

**The effect doesn‘t work because the body content hasn't been loaded when your JavaScript runs.**

By putting code inside `$(document).ready(function () {})`, you are forcing your script execution to wait until the document's content is loaded... which is critical if you’re code interacts with that content (like the ScrollReveal library).

However, this also means your page's content has to _wait_ for scripts and whatever else is in the `<head>` to load, which can cause a blank screen while your page is loading on slower connections. Many developers will recommend putting scripts just before the closing `</body>` tag for this reason, which removes the need to use jQuery's document ready helper function.

**TL;DR**

Either:
- Put your `script.js` after your body content just before the closing `</body>` tag, or...
- Move the `reveal()` call inside your `$(document).ready(function() {})`

**Comment by jlmakes** [comment]
If ScrollReveal is successfully hiding your elements @bbmattieu9, that means a fair amount is already working—but your animations aren't being triggered. With v3.x for example, one cause could be the `<body>` not having `height: 100%` (which is done for you in v4).

If you're still having problems @bbmattieu9 please open a new issue with details about your project, and ideally a recreation of the problem.

**Comment by itslandry** [comment]
i had the a similar problem. you should load the scrollreveal.min.js in the header.  this is my function.php in wordpress
function devwp_thelinkevent_scripts() {
	wp_enqueue_style( 'devwp_thelinkevent-style', get_stylesheet_uri() );

	wp_enqueue_style('devwp-bs-css', get_template_directory_uri() . '/dist/css/bootstrap.min.css');
	
	wp_register_style( 'devwp-fontawesome',"https://use.fontawesome.com/releases/v5.0.9/css/all.css" );
	wp_enqueue_style('devwp-fontawesome');

	wp_enqueue_style( 'swiper-css', get_template_directory_uri(). '/dist/css/swiper.min.css');

	wp_enqueue_script( 'swiper-js', get_template_directory_uri(). '/src/js/swiper.min.js', array('jquery'),'' , false);

	wp_enqueue_script('main-script', get_template_directory_uri() . '/src/js/main.js', array('jquery'),'',true);

	wp_enqueue_script('scrollreveal', get_template_directory_uri() . '/src/js/scrollreveal.min.js', array('jquery'), '', false);

	wp_register_script('popper', "https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.3/umd/popper.min.js", false,'',true);
	wp_enqueue_script('popper');
	
	wp_enqueue_script('devwp-tether', get_template_directory_uri() . '/src/js/tether.min.js', array('jquery'),'',true);
	wp_enqueue_script('devwp-bootstrap', get_template_directory_uri() . '/dist/js/bootstrap.min.js', array('jquery'),'',true);

	

	wp_enqueue_script( 'devwp_thelinkevent-navigation', get_template_directory_uri() . '/js/navigation.js', array(), '20151215', true );

	wp_enqueue_script( 'devwp_thelinkevent-skip-link-focus-fix', get_template_directory_uri() . '/js/skip-link-focus-fix.js', array(), '20151215', true );

	if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	}
}
i changed it to false meaning it is going to load in the header instead of footer and it worked.

**Comment by precious-adeyinka** [comment]
Hello, @pepo972 did you finally figure it out, I am having this same issue, everything u were typing would have been me typing, but the only sad thing is I can't see anything like, 'yaay' it's working now... :disappointed:

**Comment by jlmakes** [comment]
@precious-adeyinka The problem outlined in this issue was about source order — making sure that your script (and the DOM) has loaded before you try to use it (and manipulate the DOM).
